api/envoy/extensions/filters/http/local_ratelimit/v3/local_rate_limit.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.local_ratelimit.v3;
+
+import "envoy/config/core/v3/base.proto";
+import "envoy/type/v3/http_status.proto";
+import "envoy/type/v3/token_bucket.proto";
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.local_ratelimit.v3";
+option java_outer_classname = "LocalRateLimitProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Local Rate limit]
+// Local Rate limit :ref:`configuration overview <config_http_filters_local_rate_limit>`.
+// [#extension: envoy.filters.http.local_ratelimit]
+
+// [#next-free-field: 11]
+message LocalRateLimit {
+  option (udpa.annotations.versioning).previous_message_type =
+      "envoy.config.filter.http.local_rate_limit.v2.LocalRateLimit";
+
+  // This field allows to send a HTTP response status code to the downstream client other
+  // than 429 (TooManyRequests) when the request has been rate limited.
+  type.v3.HttpStatus status = 1;
```

Comment:
It's a possible scenario that the same rate limiter is applied to both grpc and http traffic (e.g. in case of transcoding or virtual_host-level rate limiter).
In this scenarios the status code options would conflict (as gRPC mandates the usage of 200 code).

Should grpc be introduced as first class citizen or are there other options?

Reply:
You mean, a same route that does both HTTP and gRPC?

Reply:
Yes, few examples:
1. There is a single gRPC route `/service.Service/Method` but it has `envoy.filters.http.grpc_json_transcoder` filter enabled. It allows the same route to serve both HTTP and gRPC traffic (all of it is converted to gRPC before going to upstream). Transcoding itself is potentially pricey (it requires buffering the whole request and is ≈reflection based), so one may want to put rate limiter before it.
2. Rate limiter is set for the whole `VirtualHost`, virtual host can serve traffic for different variations of traffic, e.g. it can have few gRPC and few HTTP upstreams.

Reply:
So we'd want to be able to configure an HTTP _and_ a gRPC status code -- OK. Do we do this anywhere else in the API?

Reply:
Not anywhere today, though see https://github.com/envoyproxy/envoy/pull/10841 where we are adding gRPC fault codes which is somewhat similar.

Reply:
Sorry, Whether the `status` field could be set to universal? Then the `rate-limit` filter module convert status-code depends on the requeste protocol.  Here is a discuss  https://github.com/envoyproxy/envoy/issues/10813 .